tool/net/redbean.c

```diff
@@ -4471,6 +4471,16 @@ static int LuaSha512(lua_State *L) {
   return LuaHasher(L, 64, mbedtls_sha512_ret_512);
 }
 
+static int LuaGetRandomBytes(lua_State *L) {
+  char *p;
+  size_t n = luaL_optinteger(L, 1, 16);
+  p = malloc(n);
+  CHECK_EQ(n, getrandom(p, n, 0));
```

Comment:
getrandom() won't return more than 256 bytes. If the requested amount is larger then you could provide cheap random using `rngset(buf, sizeof(buf), rand64, -1);`.

Reply:
Good point; added a check for this. I don't think anything longer 256 would be needed.